inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1100,6 +1100,32 @@ void InsertConcatAligningFilterPass::run() {
         auto concatLayer = info.as<ConcatLayer*>();
         IE_ASSERT(concatLayer != nullptr);
 
+        // Check if this concat has multiple inputs coming from the same layer
```

Comment:
It will be too late, the next pass will already detect a loop and fire an exception.

Reply:
Is it possible to move  InsertCopyLayerPass before the pass where the problem occurs?
If not could you please move it into a separate pass? It's more clear when every pass makes some certain work reflecting in its name.

Reply:
I have the similar fix in PR 5623 but in my case this fix was written for the problem which is reproduced differently, so I added the fix in InsertCopyLayerPass and it was enough for its fixing. So we need to be attentive before merging this fix to master to avoid the double fix and have a fix which covers the both cases. It'd be good to have a test for your specific case.

Reply:
In such case I need to create a separate ticket for fixing the loop issue + tests and only after having the PR accepted and merged I can come back to this transform.

Reply:
I pushed code from PR 5623 with minor adaptation.